[AArch64] Remove wrong processor feature

tomershafir opened the pull request:
`fmov dX, dY` is not a preferred instruction.

Previously introduced by: https://github.com/llvm/llvm-project/pull/144152

DavidSpickett commented:
No opinion on the change itself but your PR description does need more detail. A link to the change that added it would be good too.

tomershafir commented:
Done

@jroelofs Ping

davemgreen commented on llvm/test/CodeGen/AArch64/arm64-zero-cycle-regmov-fpr.ll:
```diff
@@ -1,103 +0,0 @@
-; RUN: llc < %s -mtriple=arm64-linux-gnu | FileCheck %s -check-prefixes=NOTCPU-LINUX --match-full-lines
```
Removing this loses all testing for the feature. Can we keep just the testing for default and +zcm-fpr64?

davemgreen reviewed (commented):
This sounds OK to me if it how the hardware works and someone from Apple agrees. It might be worth keeping the testing though.

tomershafir committed:
[AArch64] Remove wrong processor feature

`fmov dX, dY` is not a preferred instruction, in contrast to the documentation.

Previously introduced by: #144152

jroelofs reviewed (approved):
> This sounds OK to me if it how the hardware works and someone from Apple agrees.

Tomer also works for Apple, FWIW. Same team, different timezone.
